packages/bson-transpilers/test/yaml/partial.yaml

```diff
@@ -225,6 +243,7 @@ tests:
           java: ObjectId.isValid
           csharp: new ObjectId
           shell: new ObjectId
+          php: ''
```

Comment:
Working with that. Anyway I wrote PHP version of validation function:
````
    ObjectIdIsValidTemplate: &ObjectIdIsValidTemplate !!js/function >
        (lhs) => {
            return ``;
        }
    ObjectIdIsValidArgsTemplate: &ObjectIdIsValidArgsTemplate !!js/function >
        (lhs, arg) => {
            const singleStringify = (str) => {
                let newStr = str;
                if (
                    (str.charAt(0) === '\'' && str.charAt(str.length - 1) === '\'') ||
                    (str.charAt(0) === '"' && str.charAt(str.length - 1) === '"')) {
                    newStr = str.substr(1, str.length - 2);
                }
                return `'${newStr.replace(/\\([\s\S])|(')/g, '\\$1$2')}'`;
            }
        
            this.declarations.addFunc([]
                .concat('$isValidBSONObjectId = function ($objectId) {')
                .concat('    $valid = true;')
                .concat('    try {')
                .concat('        $obj = new ObjectId($objectId);')
                .concat('    } catch (InvalidArgumentException $e) {')
                .concat('        $valid = false;')
                .concat('    }')
                .concat('    return $valid;')
                .concat('};')
                .join('\n'));
        
            return `$isValidBSONObjectId(${singleStringify(arg)})`;
        }
````
The first template is empty and that's why for this partial test we've got empty result. Whole logic is in the second template.

Reply:
Note: in https://github.com/mongodb-js/compass/pull/3164/files#r898385603 I provided a more concise example of ObjectID validation. I propose we use that instead of this more verbose alternative.

Reply:
I'm not sure why the current diff here shows an empty string. Shouldn't it report the code from https://github.com/mongodb-js/compass/pull/3164/files#r898385603?

Reply:
If you are asking about, why this test returns empty result - that's because of this is a "partial" test. this is the result of only one of two templates. The new version of ObjectIdIsValid is:
````
    ObjectIdIsValidTemplate: &ObjectIdIsValidTemplate !!js/function >
        (lhs) => {
            return ``;
        }
    ObjectIdIsValidArgsTemplate: &ObjectIdIsValidArgsTemplate !!js/function >
        (lhs, arg) => {
            return `(\\ctype_xdigit(${this.utils.stringifyWithSingleQuotes(arg)}) && \\strlen(${this.utils.stringifyWithSingleQuotes(arg)}) == 24)`;
        }
````
So here you can see, that all logic is in the second template, while the first one is empty. That's why this test has an empty result. 

For example, here is another test, which demonstrate new logic is working:
````
    - input:
          javascript: ObjectId.isValid('5ab901c29ee65f5c8550c5b9')
          python: ObjectId.is_valid('5ab901c29ee65f5c8550c5b9')
      output:
          shell: new ObjectId('5ab901c29ee65f5c8550c5b9')
          java: ObjectId.isValid("5ab901c29ee65f5c8550c5b9")
          csharp: new ObjectId("5ab901c29ee65f5c8550c5b9")
          python: ObjectId.is_valid('5ab901c29ee65f5c8550c5b9')
          object: 'true'
          php: (\ctype_xdigit('5ab901c29ee65f5c8550c5b9') && \strlen('5ab901c29ee65f5c8550c5b9') == 24)
          ruby: BSON::ObjectId.legal?('5ab901c29ee65f5c8550c5b9')
          go: primitive.IsValidObjectID("5ab901c29ee65f5c8550c5b9")
````

Reply:
I see now in `packages/bson-transpilers/symbols/php/templates.yaml` that ObjectIdIsValidTemplate is empty and ObjectIdIsValidArgsTemplate has the one-liner code I was requesting.

I'm not able to load https://github.com/mongodb-js/compass/pull/3164/files#r898385603, and that's likely because it's a resolved thread, which GitHub unfortunately doesn't expand when direct linking.

OK to resolve.